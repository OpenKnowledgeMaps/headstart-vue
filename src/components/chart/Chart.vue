<template src="./chart.html"></template>

<script>
  export default {
    name: "Chart",
    props: {
      height: Number,
      width: Number
    }
  }
</script>


<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
    svg {
        border: 1px #000 solid;
    }
</style>
