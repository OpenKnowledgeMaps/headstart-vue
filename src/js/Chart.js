export default {
  name: 'Chart',
  props: {
    height: Number,
    width: Number,
  },
};
