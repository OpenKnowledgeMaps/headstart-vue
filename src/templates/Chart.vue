<script src="../js/Chart.js"></script>

<template>
<div class="headstart">
    <div class="container-headstart">
        <div class="vis-col">
            <div id="headstart-chart">
                <svg id="chart-svg" :height="height" :width="width">
                    <g id="chart_canvas">
                        <rect
                                id="map-rect"
                                :height="height"
                                :width="width"
                                class="zoomed_out"
                                x="0"
                                y="0"
                        ></rect>
                        <slot></slot>
                    </g>
                </svg>
            </div>
        </div>
    </div>
</div>
</template>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
    svg {
        border: 1px #000 solid;
    }
</style>
