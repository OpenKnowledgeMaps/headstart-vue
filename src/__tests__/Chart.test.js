import Chart from '../components/Chart';
import { shallowMount, mount } from '@vue/test-utils';

const factory = (values = {}) => {
  return shallowMount(Chart, {
    propsData: { ...values}
  })
};

describe('Chart', () => {
  it('matches a previously made snapshot', () => {
    const wrapper = factory({height: 600, width: 600});
    expect(wrapper).toMatchSnapshot();
  });
});