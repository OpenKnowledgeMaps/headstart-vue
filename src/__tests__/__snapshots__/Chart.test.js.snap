// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart matches a previously made snapshot 1`] = `
<div class="headstart">
  <div class="container-headstart">
    <div class="vis-col">
      <div id="headstart-chart">
        <svg id="chart-svg" height="600" width="600">
          <g id="chart_canvas">
            <rect id="map-rect" height="600" width="600" x="0" y="0" class="zoomed_out"></rect>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
